### PR TITLE
Upgrade to sbt 1.0

### PR DESCRIPTION
**Pull request by jsolmon**
* Upgrade to SBT 1.0
* Upgrade sbt-s3 to latest version (1.5.3)

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -37,11 +39,17 @@ bintrayOrganization := Some("localytics")
...
-// http://www.scala-sbt.org/0.13/docs/sbt-0.13-Tech-Previews.html#Circular+dependency
+// http://www.scala-sbt.org/1.x/docs/sbt-0.13-Tech-Previews.html#Circular+dependency
 updateOptions := updateOptions.value.withCircularDependencyLevel(CircularDependencyLevel.Error)
 
-libraryDependencies += "org.scalatest" %% "scalatest" % "2.2.6" % "test"
+libraryDependencies +=
+  ("org.scalatest" %% "scalatest" % "3.0.4" % "test"
+    // Scalatest 3.0.4 is slightly behind sbt and Scala dependencies for these two modules, so they must be excluded
+    // while Strict dependency checking is enabled.
+    exclude("org.scala-lang.modules", s"scala-xml_${scalaBinaryVersion.value}")
```
Comment (prurph) [inline build.sbt:53]: This is a dumb question but can you use the `%%` syntax here? Like

`exclude("org.scala-lang.modules" %% "scala-xml")`

or is that just a thing that would make sense in a world where sbt makes sense?

Comment (jsolmon) [reply build.sbt:53]: I gave this a try but sadly it does not compile.

Comment (prurph) [reply build.sbt:53]: Ah, my bad, it's `excludeAll("org.scala-lang.modules" %% "scala-xml")`

Comment (prurph) [reply build.sbt:53]: (the above compiles--I'm assuming if it compiles it works: that's how it goes with Scala, right? 😁 )

**Review by prurph — APPROVED** [review]
I'm fine with this either way--but if excludeAll does fit the bill it would be cool to use it since sbt patterns tend to propagate themselves throughout the codebase.

**Pull request merged**